core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +86,152 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
+        Comparator<Offer> offerPriceComparator = Comparator.comparing(Offer::getPrice, Comparator.nullsLast(Comparator.naturalOrder()));
+
+        // Trading btc-fiat is considered as buying/selling BTC, but trading btc-altcoin is
+        // considered as buying/selling Altcoin. Because of this, when viewing a btc-altcoin pair,
+        // the buy column is actually the sell column and vice versa. To maintain the expected
+        // ordering, we have to reverse the price comparator.
+        boolean isCrypto = CurrencyUtil.isCryptoCurrency(currencyCode);
+        if (isCrypto) offerPriceComparator = offerPriceComparator.reversed();
+
+        // Offer amounts are used for the secondary sort. They are sorted from high to low.
+        Comparator<Offer> offerAmountComparator = Comparator.comparing(Offer::getAmount).reversed();
+
+        var buyOfferSortComparator =
+                offerPriceComparator.reversed() // Buy offers, as opposed to sell offers, are primarily sorted from high price to low.
+                        .thenComparing(offerAmountComparator);
+        var sellOfferSortComparator =
+                offerPriceComparator
+                        .thenComparing(offerAmountComparator);
+
+        List<Offer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList()); 
+        List<Offer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList());
+
+        final Optional<Offer> highestBuyPriceOffer = buyOffers.stream()
+                .filter(o -> o.getPrice() != null)
+                .max(Comparator.comparingLong(o -> o.getPrice().getValue()));
+
+        if (highestBuyPriceOffer.isPresent()) {
+            final Offer offer = highestBuyPriceOffer.get();
+        } else {
+            log.debug("highestBuyPriceOffer not present");
+        }
+
+        final Optional<Offer> highestBuyVolumeOffer = buyOffers.stream()
+                .filter(o -> o.getVolume() != null)
+                .max(Comparator.comparingLong(o -> o.getVolume().getValue()));
+
+        if (highestBuyVolumeOffer.isPresent()) {
+            final Offer offer = highestBuyVolumeOffer.get();
+        }
+
+        //DEBUG buildChartAndTableEntries called
+        double accumulatedAmount = 0;
+        
+        // Create buyer hashmap {key:price, value:count}, uses TreeMap to sort by key (asc)
+        TreeMap<Double,Double> buyTM = new TreeMap<Double,Double>();        
+        for(Offer offer: buyOffers) {            
+            // Double price = offer.getPrice().getDoubleValue() / offer.getOfferPayload().getCoinAmount();//Price per XMR coin;
+            // Long previousValue = buyTM.get(price) != null ? buyTM.get(price) : Long.valueOf(0);
```

Comment:
please remove the commented out code if it's not needed

Reply:
done